Analytics.php

```diff
@@ -338,11 +338,11 @@ public function hasPageViewQueue()
     }
 
     /**
-     * @return Symfony\Component\HttpFoundation\Request $request
+     * @return Symfony\Component\HttpFoundation\Request|null $request
      */
     public function getRequest()
     {
-        return $this->container->get('request');
+        return $this->container->get('request_stack')->getCurrentRequest();
```

Comment:
i think you should probably update `composer.json` as well to require Symfony >= 3.x

Reply:
Yes request_stack is available in 3. In general symfony doesn't ever introduce new features in .0 releases - they just remove deprecated stuff. 

See https://symfony.com/blog/new-in-symfony-2-4-the-request-stack
"As of Symfony 2.4, the best practice is to never inject the request service, but to inject the request_stack service instead" 

So I think what I should do is update composer.json to require Symfony >= 2.4